addons/purchase/models/purchase.py

```diff
@@ -706,25 +708,45 @@ def _confirm_reception_mail(self):
                 order.message_post(body="The order receipt has been acknowledged by %s." % order.partner_id.name)
 
     def _update_date_planned_for_lines(self, updated_dates):
-        if updated_dates:
-            note = self._compose_note(updated_dates)
-            self.activity_schedule(
+        # creat or update the activity
+        self._create_update_date_activity(updated_dates)
+        # update the date on PO line
+        for line, date in updated_dates:
+            line._update_date_planned(date)
+
+    def _create_update_date_activity(self, updated_dates):
+        """Helper method for creating or updating an activity for the user when
+        vendor update scheduled date on portal website."""
+        activity = self.env['mail.activity'].search([
+            ('summary', '=', 'Date Updated'),
+            ('res_model_id', '=', self.env['ir.model']._get(self._name).id),
+            ('res_id', '=', self.id),
+            ('user_id', '=', self.user_id.id)], limit=1)
+        if activity:
+            for line, date in updated_dates:
+                # replace special characters in product_name by '.'
+                product_name_re = re.sub('[\[\]\(\)\@\$\?\{\}\|\*\.\+\/\\\]', '.', line.product_id.display_name)
+                pattern = re.compile('<p> \\\xa0 - %s.*?</p>' % product_name_re)
+                new_line = _('<p> &nbsp; - %s from %s to %s </p>') % (line.product_id.display_name, line.date_planned.date(), date.date())
+                if re.search(pattern, activity.note):
+                    activity.note = re.sub(pattern, new_line, activity.note)
+                else:
+                    # here we insert it before any other product instead of simply
+                    # appending it to the end because in purchase_stock picking
+                    # info will be the last line. This way we won't append it after
+                    # the picking info.
+                    activity.note = re.sub('</p><p>', '</p>%s<p>' % new_line, activity.note, count=1)
```

Comment:
is it really important to update the text? only having a text with the new date would simplify this logic
maybe having the scheduled date tracked on the picking would be enough ?